template.go

```diff
@@ -32,9 +30,76 @@ type Template struct {
 	Invocation Invocation `yaml:"invocation,omitempty"`
 }
 
+// template
+type template struct {
```

Comment:
If you do take my suggestion and keep `template` only, removing `templateConfig`, you would just need to capitalize those fields you want serialized to disk (BuildConfig, Invocation)

Reply:
see [comment](https://github.com/knative-sandbox/kn-plugin-func/pull/961#discussion_r849818376)